salesagility/SuiteCRM
# Fix scrm 638

## Description



Ensures that a user can only send an email via their own inbound email account.


## Motivation and Context

* Adds a check to the send action in the email controller.
* It verifies that they user is subscribed to the inbound email in question
* It verifies the the from address is in the users inbound email account or in the system settings. 

## How To Test This

* Repair and rebuild
* Set up a user with a personal email account
* Take a note of the inbound email id and the from address
* Set up a second user
* Login in to the second user account 
* set the url  to http://path-to-suitecrm-instance/index.php?module=Emails&action=Send&inbound_email_record=[noted inbound email record id]&from_addr=[noted from email address]

You should see and error message, and when you refresh or access the suitecrm again it should ask you to login again.

## Types of changes


### Final checklist
